### PR TITLE
Add specs for BsRequestActivityTimelineComponent

### PR DESCRIPTION
**Pull request by saraycp**
We had implemented the component but not the specs. Here they are.

### CODE REVIEW

File: src/api/spec/components/bs_request_activity_timeline_component_spec.rb [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
+    it 'shows the comments threads' do
+      expect(render_inline(described_class.new(bs_request: bs_request))).to have_text('accepted request')
+    end
+
+    it 'shows the history elements' do
+      expect(render_inline(described_class.new(bs_request: bs_request))).to have_text('wrote')
+      expect(render_inline(described_class.new(bs_request: bs_request))).to have_text('1 day ago')
+    end
+
+    it 'shows the comment first, as it is older timeline item' do
```
Comment (danidoni) [inline src/api/spec/components/bs_request_activity_timeline_component_spec.rb:24]:
```suggestion
    it 'shows the comment first, as it is an older timeline item' do
```

File: src/api/spec/components/bs_request_activity_timeline_component_spec.rb [hunk 2]
```diff
@@ -1,13 +1,24 @@
 require 'rails_helper'
 
 RSpec.describe BsRequestActivityTimelineComponent, type: :component do
-  context 'when we provide a bs_request' do
-    it 'shows an element telling who created the request and when'
-    it 'shows the comments threads'
-    it 'shows the history elements'
+  context 'when we do not provide a bs_request' do
+    it do
+      expect { render_inline(described_class.new) }.to raise_error(ArgumentError, 'missing keyword: :bs_request')
```
Comment (dmarcoux) [inline src/api/spec/components/bs_request_activity_timeline_component_spec.rb:6]: I don't really see the benefit in testing how keyword arguments work. This is a core Ruby feature. I suggest removing this test.

Comment (saraycp) [reply src/api/spec/components/bs_request_activity_timeline_component_spec.rb]: You are right. Changes pushed. I'm going to delete it from the other specs I added too.

**Comment by danidoni** [comment]
Don't we test the presence of the avatar and the creation date?

**Comment by saraycp** [comment]
> Don't we test the presence of the avatar and the creation date?

I don't think it should be part of this spec, as this component is in charge of displaying a comment or a history element in the right order. That's why I check the minimum info possible to simply know if a comment or a history element is displayed and in the right order.

Checking the details of the comments and history elements should be done in the specific components' specs. As I already did with the icon [in the specs for the history element](https://github.com/openSUSE/open-build-service/blob/bce6593e23921f66f54f8aedeb2fa15e882edf38/src/api/spec/components/bs_request_history_element_component_spec.rb#L17) I could create new PRs to check for more details...

**Comment by danidoni** [comment]
> > Don't we test the presence of the avatar and the creation date?
> 
> I don't think it should be part of this spec, as this component is in charge of displaying a comment or a history element in the right order. That's why I check the minimum info possible to simply know if a comment or a history element is displayed and in the right order.

Okay!

**Review by danidoni — APPROVED** [review]

**Pull request merged**